tests/psa-client-server/psasim/src/psa_ff_server.c

```diff
@@ -133,7 +133,7 @@ size_t psa_write(psa_handle_t msg_handle, uint32_t outvec_idx, const void *buffe
 
     if (atomic_load(&(shared_memory->owner)) != SHARED_MEMORY_OWNER_SERVER) {
         ERROR("Cannot write on shared memory while it's owned by the client");
-        return PSA_ERROR_BAD_STATE;
```

Comment:
The documentation of this function is now:
```
 * \retval >0                   On success the number of written bytes is returned.
 * \retval 0                    On failure, which can happen if:
 *                              - outvec_idx is invalid, i.e. outvec_idx > PSA_MAX_IOVEC,
 *                              - the server tries to write a message when it's not
 *                                owning the shared memory.
 ```
so I think that returning 0 in case of errors makes sense. Moreover the return type is `size_t`, which is unsigned so returning a signed value might "erroneously" appear as a large positive value for anyone checking the return value. I know that it would be unpractical as value since it would be very large, but still...